sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -812,6 +812,9 @@ size_t coap_opt_put_uri_pathquery(uint8_t *buf, uint16_t *lastonum, const char *
     size_t bytes_out = coap_opt_put_string_with_len(buf, *lastonum,
                                                     COAP_OPT_URI_PATH,
                                                     uri, len, '/');
+    if (bytes_out == 0) {
+        return 0;
```

Comment:
Why? `coap://host?query=foo` is a legal URI that'd have a `len == 0` here, would thus AIU produce `bytes_out == 0`, but still needs continuation to put query options.

Reply:
oops, I overlooked the `query` part, should be fixed by d1204d557e57332496a46f7524bc548ae845e7e5.